imap_processing/lo/l1c/lo_l1c.py

```diff
@@ -0,0 +1,171 @@
+            data_field_tup("TRIPLES_COUNTS"),
+            data_field_tup("TRIPLES_RATES"),
+            data_field_tup("DOUBLES_COUNTS"),
+            data_field_tup("DOUBLES_RATES"),
+            data_field_tup("HYDROGEN_COUNTS"),
+            data_field_tup("HYDROGEN_RATES"),
+            data_field_tup("OXYGEN_COUNTS"),
+            data_field_tup("OXYGEN_RATES"),
+            data_field_tup("EXPOSURE_TIME"),
+        ]
+
+    dataset = create_datasets(attr_mgr, logical_source, data_fields)
+    create_file_paths = write_cdf(dataset)
+    return create_file_paths
+
+
+# TODO: This is going to work differently when I sample data.
+#  The data_fields input is temporary.
+def create_datasets(attr_mgr, logical_source, data_fields):
+    """Create a dataset using the populated data classes.
+
+    Parameters
+    ----------
+    attr_mgr : ImapCdfAttributes
+        attribute manager used to get the data product field's attributes
+    logical_source: str
+        The logical source of the data product that's being created
+    data_fields: list[dataclasses.Field]
+
+    Returns
+    -------
+    xr.dataset
+        dataset with all data product fields in xr.DataArray
+    """
+    # TODO: Once L1B DE processing is implemented using the spin packet
+    #  and relative L1A DE time to calculate the absolute DE time,
+    #  this epoch conversion will go away and the time in the DE dataclass
+    #  can be used direction
+    epoch_converted_time = [calc_start_time(1)]
+
+    # Create a data array for the epoch time
+    # TODO: might need to update the attrs to use new YAML file
+    epoch_time = xr.DataArray(
+        data=epoch_converted_time,
+        name="epoch",
+        dims=["epoch"],
+        attrs=attr_mgr.get_variable_attributes("epoch"),
+    )
+
+    if logical_source == "imap_lo_l1c_pset":
+        esa_step = xr.DataArray(
+            data=[1, 2, 3, 4, 5, 6, 7],
+            name="esa_step",
+            dims=["esa_step"],
+            attrs=attr_mgr.get_variable_attributes("esa_step"),
+        )
+        pointing_bins = xr.DataArray(
+            data=np.arange(3600),
+            name="pointing_bins",
+            dims=["pointing_bins"],
+            attrs=attr_mgr.get_variable_attributes("pointing_bins"),
+        )
+
+        esa_step_label = xr.DataArray(
+            esa_step.values.astype(str),
+            name="esa_step_label",
+            dims=["esa_step_label"],
+            attrs=attr_mgr.get_variable_attributes("esa_step_label"),
+        )
+        pointing_bins_label = xr.DataArray(
+            pointing_bins.values.astype(str),
+            name="pointing_bins_label",
+            dims=["pointing_bins_label"],
+            attrs=attr_mgr.get_variable_attributes("pointing_bins_label"),
+        )
+        dataset = xr.Dataset(
+            coords={
+                "epoch": epoch_time,
+                "pointing_bins": pointing_bins,
+                "pointing_bins_label": pointing_bins_label,
```

Comment:
That's a good question. If they can be normal variables, I feel like it still makes more sense to have them as coordinates since they're basically string versions of the coordinates. I haven't looked at it much yet, but I'm hoping there is a good way to just convert the data to a string and use that as the label within write_cdf or cdflib instead of having to define attributes for the label since all the attributes are easily determined (ie. string length) or straight from the coordinate attributes.